source/extensions/common/wasm/wasm.cc

```diff
@@ -470,6 +400,56 @@ uint64_t getMetricHandler(void* raw_context, uint32_t metric_id) {
   return context->getMetric(metric_id);
 }
 
+uint32_t grpcCallHandler(void* raw_context, uint32_t service_ptr, uint32_t service_size,
+                         uint32_t service_name_ptr, uint32_t service_name_size,
+                         uint32_t method_name_ptr, uint32_t method_name_size, uint32_t request_ptr,
+                         uint32_t request_size, uint32_t timeout_milliseconds) {
+  auto context = WASM_CONTEXT(raw_context);
+  auto service = context->wasmVm()->getMemory(service_ptr, service_size);
+  auto service_name = context->wasmVm()->getMemory(service_name_ptr, service_name_size);
+  auto method_name = context->wasmVm()->getMemory(method_name_ptr, method_name_size);
+  auto request = context->wasmVm()->getMemory(request_ptr, request_size);
+  envoy::api::v2::core::GrpcService service_proto;
+  if (!service_proto.ParseFromArray(service.data(), service.size())) {
+    return false;
+  }
+  return context->grpcCall(service_proto, service_name, method_name, request,
+                           std::chrono::milliseconds(timeout_milliseconds));
+}
+
+uint32_t grpcStreamHandler(void* raw_context, uint32_t service_ptr, uint32_t service_size,
+                           uint32_t service_name_ptr, uint32_t service_name_size,
+                           uint32_t method_name_ptr, uint32_t method_name_size) {
+  auto context = WASM_CONTEXT(raw_context);
+  auto service = context->wasmVm()->getMemory(service_ptr, service_size);
+  auto service_name = context->wasmVm()->getMemory(service_name_ptr, service_name_size);
+  auto method_name = context->wasmVm()->getMemory(method_name_ptr, method_name_size);
+  envoy::api::v2::core::GrpcService service_proto;
+  if (!service_proto.ParseFromArray(service.data(), service.size())) {
+    return false;
+  }
+  return context->grpcStream(service_proto, service_name, method_name);
+}
+
+void grpcCancelHandler(void* raw_context, uint32_t token) {
+  auto context = WASM_CONTEXT(raw_context);
+  context->grpcCancel(token);
+}
+
+void grpcCloseHandler(void* raw_context, uint32_t token) {
+  auto context = WASM_CONTEXT(raw_context);
+  context->grpcClose(token);
+}
+
+void grpcSendHandler(void* raw_context, uint32_t token, uint32_t message_ptr, uint32_t message_size,
+                     uint32_t end_stream) {
+  auto context = WASM_CONTEXT(raw_context);
+  auto message = context->wasmVm()->getMemory(message_ptr, message_size);
+  context->grpcSend(token, message, end_stream);
+}
+
+// uint32_t getTotalMemoryHandler(void*) { return 0x7FFFFFFF; }
```

Comment:
This can be removed, it was removed from Emscripten few releases ago.

Reply:
Done